src/library/columncache.cpp

```diff
@@ -5,9 +5,13 @@
 #include "library/dao/cratedao.h"
 
 void ColumnCache::setColumns(const QStringList& columns) {
+    m_columnsByIndex.clear();
```

Comment:
We get the truth by a QStringList here and the line below is the only place we write the coloums to m_columnByIndex exactly as they are. So there is no way to mess up m_columnByIndex. The old loop filling the QMap was just redundant and the original point of confusion. 

Now we set up m_columnIndexByName, by iterating over columns[i] which is equal to m_columnByIndex.
m_columnIndexByName is used to setup m_columnIndexByEnum via fieldIndex().

So we have a reliable round trip.